Add matrix integration node

krynble opened the pull request:
Created node for integrating to https://matrix.org/

Resources and operations:
- Account
Who am I
- Event
Get event by ID
- Message
Read
Send
- Room
Create a room
Join a room
Invite
Kick
Leave
List members
- Sync
Read

For reference the documentation can be found here: https://matrix.org/docs/api/client-server/

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=1024) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=1024) before we can accept your contribution.<br/><hr/>**Omar Ajoue** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=1024) it.</sub>

krynble committed:
Added Matrix integration node

krynble committed:
fix: Improve code quality and add new operation

- Changed operation names to match casing (all camelCase)
- Ordering operation names to be alphabetical
- Creating a read all operation to fetch all messages from a room
- Added node subtitle

krynble committed:
fix: add element index so that expressions work on multiple items

krynble committed:
feature: added possibility to upload and send media to Matrix

- Also replacing Promises.all() + Array.map() For a regular for as it messes up ordering

krynble committed:
refactor: merging upload + send Media in a single action

krynble committed:
refactor: improved code quality and endpoints

- Removed sync entirely as a better way to retrieve messages is now
implemeented
- Added rooms dropdown to operations
- Added option to paginate or retrieve all room messages
- Removed option to upload media from text contents. Only files are
accepted now
- Room members has bem moved from the Rooms resource to a standalone
with Get All operation

krynble committed:
Added filters to get messages

janober commented:
@krynble You sadly made changes on the wrong PR. @RicardoE105 already made some improvements over at #1045. Can you please add them there again. Thanks!

janober commented:
Ignore the above message. Got merged with #1046

janober commented:
Got released with n8n@0.87.0
